concordium-consensus/src/Concordium/Queries.hs

```diff
@@ -235,107 +320,176 @@ liftSkovQueryBHIAndVersion query bhi = do
+        bb <- bestBlock
+        let csBestBlock = getHash bb
+        let csBestBlockHeight = absoluteHeight bb
+        csCurrentEraGenesisBlock <- getHash <$> genesisBlock
+        csCurrentEraGenesisTime <- timestampToUTCTime <$> getGenesisTime
+        genData <- getGenesisData
+        let csSlotDuration = gdSlotDuration genData
+        let csEpochDuration = fromIntegral (gdEpochLength genData) * csSlotDuration
+        lfb <- lastFinalizedBlock
+        let csLastFinalizedBlock = getHash lfb
+        let csLastFinalizedBlockHeight = absoluteHeight lfb
+        let csGenesisIndex = evcIndex evc
+        let csProtocolVersion = demoteProtocolVersion (protocolVersion @pv)
+        stats <- getConsensusStatistics
+        let csBlocksReceivedCount = stats ^. blocksReceivedCount
+            csBlockLastReceivedTime = stats ^. blockLastReceived
+            csBlockReceiveLatencyEMA = stats ^. blockReceiveLatencyEMA
+            csBlockReceiveLatencyEMSD = sqrt $ stats ^. blockReceiveLatencyEMVar
+            csBlockReceivePeriodEMA = stats ^. blockReceivePeriodEMA
+            csBlockReceivePeriodEMSD = sqrt <$> stats ^. blockReceivePeriodEMVar
+            csBlocksVerifiedCount = stats ^. blocksVerifiedCount
+            csBlockLastArrivedTime = stats ^. blockLastArrive
+            csBlockArriveLatencyEMA = stats ^. blockArriveLatencyEMA
+            csBlockArriveLatencyEMSD = sqrt $ stats ^. blockArriveLatencyEMVar
+            csBlockArrivePeriodEMA = stats ^. blockArrivePeriodEMA
+            csBlockArrivePeriodEMSD = sqrt <$> stats ^. blockArrivePeriodEMVar
+            csTransactionsPerBlockEMA = stats ^. transactionsPerBlockEMA
+            csTransactionsPerBlockEMSD = sqrt $ stats ^. transactionsPerBlockEMVar
+            csFinalizationCount = stats ^. finalizationCount
+            csLastFinalizedTime = stats ^. lastFinalizedTime
+            csFinalizationPeriodEMA = stats ^. finalizationPeriodEMA
+            csFinalizationPeriodEMSD = sqrt <$> stats ^. finalizationPeriodEMVar
+        return ConsensusStatus{..}
+    statusV1 ::
+        forall (pv :: ProtocolVersion).
+        (IsProtocolVersion pv) =>
+        (BlockHash, UTCTime) ->
+        EVersionedConfiguration finconf ->
+        VersionedSkovV1M finconf pv ConsensusStatus
+    statusV1 (csGenesisBlock, csGenesisTime) evc = do
+        let absoluteHeight = localToAbsoluteBlockHeight (evcGenesisHeight evc) . SkovV1.blockHeight
+        bb <- use SkovV1.lastFinalized -- FIXME: Use best block.
+        let csBestBlock = getHash bb
+        let csBestBlockHeight = absoluteHeight bb
+
+        genMetadata <- use SkovV1.genesisMetadata
+        let csCurrentEraGenesisBlock = SkovV1.gmCurrentGenesisHash genMetadata
+        let csCurrentEraGenesisTime =
+                timestampToUTCTime $
+                    BaseV1.genesisTime $
+                        SkovV1.gmParameters genMetadata
+        let csSlotDuration = 0 -- FIXME: What to do here?
```

Comment:
Did we not agree to remove this for P6 and up?

Reply:
Deferred to issue #857